### ISSUE TITLE
Rebuild fails with pgp error/prompt (signature from ... is marginal trust)

### ISSUE THREAD

**Issue by kpcyrd** [archlinux/archlinux-repro#114]
Noticed this just now:
```
% repro -o out cargo-edit-0.10.4-1-x86_64.pkg.tar.zst
repro 20220606
==> Reusing existing container
==> Updating container
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
:: Starting full system upgrade...
resolving dependencies...
looking for conflicting packages...

Packages (82) acl-2.3.1-2  arch-install-scripts-26-1  archlinux-keyring-20220727-1  attr-2.5.1-2  audit-3.0.8-1  brotli-1.0.9-8  ca-certificates-mozilla-3.81-1  coreutils-9.1-1  cryptsetup-2.5.0-1  curl-7.84.0-2  dbus-1.14.0-1
              device-mapper-2.03.16-1  e2fsprogs-1.46.5-4  expat-2.4.8-1  file-5.42-1  findutils-4.9.0-1  gcc-libs-12.1.0-3  gdbm-1.23-1  gettext-0.21-2  glib2-2.72.3-1  glibc-2.35-6  gmp-6.2.1-2  gnupg-2.2.36-1  gnutls-3.7.6-1
              gpgme-1.17.1-1  grep-3.7-1  hwdata-0.361-1  iana-etc-20220603-1  icu-71.1-1  iptables-1:1.8.8-2  json-c-0.16-1  kbd-2.5.1-1  kmod-30-1  krb5-1.19.3-3  libarchive-3.6.1-1  libcap-2.65-1  libcap-ng-0.8.3-1  libelf-0.187-1
              libevent-2.1.12-1  libffi-3.4.2-5  libgcrypt-1.10.1-1  libgpg-error-1.45-2  libidn2-2.3.3-1  libldap-2.6.3-1  libmnl-1.0.5-1  libnetfilter_conntrack-1.0.9-1  libnfnetlink-1.0.2-1  libnftnl-1.2.2-1  libnghttp2-1.48.0-1
              libnl-3.7.0-1  libp11-kit-0.24.1-1  libpcap-1.10.1-2  libsasl-2.1.28-1  libseccomp-2.5.4-1  libsecret-0.20.5-2  libsysprof-capture-3.44.0-1  libunistring-1.0-1  libverto-0.3.2-4  libxcrypt-4.4.28-2  libxml2-2.9.14-1
              linux-api-headers-5.17.5-2  mpfr-4.1.0.p13-2  ncurses-6.3-3  nettle-3.8-1  openssl-1.1.1.q-1  p11-kit-0.24.1-1  pacman-6.0.1-7  pacman-mirrorlist-20220724-1  pcre2-10.40-1  popt-1.18-3  readline-8.1.002-1  shadow-4.11.1-1
              sqlite-3.39.1-1  systemd-251.3-1  systemd-libs-251.3-1  tpm2-tss-3.2.0-1  tzdata-2022a-1  util-linux-2.38-1  util-linux-libs-2.38-1  xz-5.2.5-3  zlib-1:1.2.12-2  zstd-1.5.2-7

Total Download Size:     0.08 MiB
Total Installed Size:  443.18 MiB
Net Upgrade Size:       73.44 MiB

:: Proceed with installation? [Y/n] 
:: Retrieving packages...
 libcap-2.65-1-x86_64                                                                                                83.6 KiB  1145 KiB/s 00:00 [########################################################################################] 100%
(82/82) checking keys in keyring                                                                                                                [########################################################################################] 100%
(82/82) checking package integrity                                                                                                              [########################################################################################] 100%
error: libcap: signature from "David Runge <dvzrv@archlinux.org>" is marginal trust
:: File /var/cache/pacman/pkg/libcap-2.65-1-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] 
error: failed to commit transaction (invalid or corrupted package)
Errors occurred, no packages were upgraded.
```

Starting from scratch currently fails because the Arch Linux release for August 2022 is not done yet, so I can't test that right now.

**Comment by vosskaem** [comment]
Total Installed Size:  3466.15 MiB
Net Upgrade Size:       240.41 MiB

:: Proceed with installation? [Y/n] y
(666/666) checking keys in keyring                                                                                     [########################################################################] 100%
(666/666) checking package integrity                                                                                   [########################################################################] 100%
error: libffi: signature from "David Runge <dvzrv@archlinux.org>" is marginal trust
:: File /var/cache/pacman/pkg/libffi-3.4.4-1-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] ^C
Interrupt signal received

**Comment by kpcyrd** [comment]
@vosskaem can you share which package you're trying to rebuild? :)

**Comment by vosskaem** [comment]
I tried to update archlinux after about 3 years. Finally I could fix it.
Normaly I always start such big update with installing archlinux-keyrings but this alone does not the job this time.
I got a lot of "Import the following key...", later I got a message that a key has marginal trust. (the message above).
After Importing keys via gpg by hand and later installing again archlinux-keyrings, it worked.
I don't know, why it was not enough to update  archlinux-keyrings first.
